core/src/test/scala/org/broadinstitute/dsde/rawls/webservice/StatusApiServiceSpec.scala

```diff
@@ -44,37 +46,37 @@ class StatusApiServiceSpec extends ApiServiceSpec with Eventually {
   ) extends ApiServices
       with MockUserInfoDirectives
 
-  def withApiServices[T](dataSource: SlickDataSource, subsystemsOk: Boolean, apiService: TestApiService)(
-    testCode: TestApiService => T
-  ): T =
+  def withApiServices[T](dataSource: SlickDataSource, apiService: TestApiService)(testCode: TestApiService => Any): Any =
     try {
-      initializeSubsystems(apiService, subsystemsOk)
+      initializeSubsystems(apiService)
       testCode(apiService)
     } finally
       apiService.cleanupSupervisor
 
-  def withConstantTestDataApiServices[T](subsystemsOk: Boolean)(testCode: TestApiService => T): T =
+  def withConstantTestDataApiServices[T](testCode: TestApiService => Any): Any =
     withConstantTestDatabase { dataSource: SlickDataSource =>
       val apiService = new TestApiService(dataSource, new MockGoogleServicesDAO("test"), new MockGooglePubSubDAO)
-      withApiServices(dataSource, subsystemsOk, apiService)(testCode)
+      withApiServices(dataSource, apiService)(testCode)
     }
 
-  def withConstantCriticalErrorTestDataApiServices[T](subsystemsOk: Boolean)(testCode: TestApiService => T): T =
+  def withConstantCriticalErrorTestDataApiServices[T](testCode: TestApiService => Any): Any =
     withConstantTestDatabase { dataSource: SlickDataSource =>
       val apiService = new TestApiService(dataSource, new MockGoogleServicesCriticalErrorDAO, new MockGooglePubSubDAO)
-      withApiServices(dataSource, subsystemsOk, apiService)(testCode)
+      withApiServices(dataSource, apiService)(testCode)
     }
 
-  def withConstantErrorTestDataApiServices[T](subsystemsOk: Boolean)(testCode: TestApiService => T): T =
+  def withConstantErrorTestDataApiServices[T](testCode: TestApiService => Any): Any =
     withConstantTestDatabase { dataSource: SlickDataSource =>
       val apiService = new TestApiService(dataSource, new MockGoogleServicesErrorDAO, new MockGooglePubSubDAO)
-      withApiServices(dataSource, subsystemsOk, apiService)(testCode)
+      withApiServices(dataSource, apiService)(testCode)
     }
 
-  def initializeSubsystems(apiService: TestApiService, subsystemsOk: Boolean) =
+  def initializeSubsystems(apiService: TestApiService): Unit = {
+    when (apiService.billingProfileManagerDAO.getStatus()) thenReturn new SystemStatus().ok(true)
```

Comment:
Why does BPM (and only BPM) need to be explicitly handled here?

Reply:
I believe we’re explicitly mocking `getStatus()` for BPM because [billingProfileManagerDAO](https://github.com/broadinstitute/rawls/blob/yp-wor-552/core/src/test/scala/org/broadinstitute/dsde/rawls/webservice/ApiServiceSpec.scala#L211) in ApiServiceSpec is using mockito (`mock[BillingProfileManagerDAO]`) instead of the manual mock classes ( [workspaceManagerDAO](https://github.com/broadinstitute/rawls/blob/yp-wor-552/core/src/test/scala/org/broadinstitute/dsde/rawls/webservice/ApiServiceSpec.scala#L170) uses the mock class)